dash/include/dash/memory/GlobUnitMem.h

```diff
@@ -187,7 +185,9 @@ class GlobUnitMem
   ~GlobUnitMem()
   {
     DASH_LOG_TRACE_VAR("GlobUnitMem.~GlobUnitMem()", _begptr);
-    _allocator.deallocate(_begptr);
+    if (_owns_mem) {
+      _allocator.deallocate(_begptr);
+    }
```

Comment:
With something like that, shouldn't the copy constructor of `GlobUnitMem` be `deleted` to avoid dangling references if one of the copies goes out of scope?